app/interactors/book_suggestions_by_genre.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class BookSuggestionsByGenre
+  include Interactor
+
+  def call
+    context.fail!(message: 'Book no has genre') unless book.genre
+    context.books = Book.where(genre: book.genre).where.not(id: book_ids)
+  end
+
+  private
+
+  def book_ids
+    user.rents.pluck(:book_id).uniq
+  end
+
+  def user
```

Comment:
You are using `user` only in `book_ids`, I don't really see the point in extracting this to a method.